contrib/dash/exsrvmonit/exsrvmonit.py

```
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import argparse
+import aiorpcx
+import asyncio
+import errno
+import encodings.idna  # noqa: F401 (need for pyinstaller build)
+import fcntl
+import json
+import os
+import socket
+import ssl
+import stat
+import sys
+import time
+from aiorpcx import RPCSession
+
+
+CLIENT_NAME = 'exsrvmonit'
+HOME_DIR = os.path.expanduser('~')
+DATA_DIR = os.path.join(HOME_DIR, f'.{CLIENT_NAME}')
+RECENT_FNAME = os.path.join(DATA_DIR, 'recent_data')
+PID_FNAME = os.path.join(DATA_DIR, f'{CLIENT_NAME}.pid')
+PID = str(os.getpid())
+MIN_PROTO_VERSION = '1.4'
+NUM_RECENT_DATA = 1440
+SERVERS_LIST = [
+    'electrum.dash.siampm.com:50002',
+    'drk.p2pay.com:50002',
+]
+
+
+def get_ssl_context():
+    sslc = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+    sslc.check_hostname = False
+    sslc.verify_mode = ssl.CERT_NONE
+    return sslc
+
+
+def peer_info_as_dict(peer):
+    res = {} 
+    res['ip'] = peer[0] 
+    res['hostname'] = peer[1] 
+    for i in peer[2]: 
+        if i.startswith('s') and len(i) > 1: 
+            res['ssl_port'] = i[1:] 
+    return res
+
+
+def read_recent_file():
+    try:
+        with open(RECENT_FNAME, 'r', encoding='utf-8') as f:
+            data = f.read()
+            return json.loads(data)
+    except Exception:
+        return []
+
+
+def save_recent_file(recent):
+    s = json.dumps(recent, indent=4, sort_keys=True)
+    try:
+        with open(RECENT_FNAME, 'w', encoding='utf-8') as f:
+            f.write(s)
+    except Exception:
+        pass
```

Same here. Logging? Somehow the situation must be processed and not just silently passed.

---

fixed in 44daed8b